projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts

```diff
@@ -169,6 +166,29 @@ export function useSyncMedia(
 
 	const isSaving = useSelect( select => select( editorStore ).isSavingPost(), [] );
 	const wasSaving = usePrevious( isSaving );
+
+	// In native, it's not currently possible to access a post's saved state from the editor store.
+	// We therefore need to listen to a native event emitter to know when a post has just been saved.
+	const [ postHasBeenJustSavedNative, setPostHasBeenJustSavedNative ] = useState( false );
+
+	useEffect( () => {
+		if ( ! isNative ) {
+			return;
+		}
+
+		const handlePostSaveEvent = () => {
+			setPostHasBeenJustSavedNative( true );
+		};
+
+		const subscription = subscribePostSaveEvent( handlePostSaveEvent );
```

Comment:
Nitpick: Since `handlePostSaveEvent` is only referenced here, I think we can simplify this with the following code.
```suggestion
		const subscription = subscribePostSaveEvent( () => setPostHasBeenJustSavedNative( true ) );
```